packages/ui/src/multiplying-architecture/KaotoEditor.tsx

```diff
@@ -59,87 +55,91 @@ export const KaotoEditor = () => {
         inset={inset.current}
         isBox
         unmountOnExit
-        activeKey={activeTabKey}
-        onSelect={handleTabClick as TabsProps['onSelect']}
+        activeKey={currentLocation.pathname}
         aria-label="Tabs in the Kaoto editor"
         role="region"
       >
         {availableTabs.design && (
-          <Tab
-            eventKey={TabList.Design}
-            className="shell__tab-content_design-canvas"
-            title={
-              <>
-                <TabTitleIcon>
-                  <Icon>
-                    <img src={camelIcon} alt="Camel icon" />
-                  </Icon>
-                </TabTitleIcon>
-                <TabTitleText>Design</TabTitleText>
-              </>
-            }
-            aria-label="Design canvas"
-          >
-            <DesignTab />
```

Comment:
No more linking components directly, the router will take care of mounting the right component when needed.